pyhap/hap_handler.py

```diff
@@ -652,14 +652,22 @@ def _handle_remove_pairing(self, tlv_objects):
 
     def handle_resource(self):
         """Get a snapshot from the camera."""
-        image_size = json.loads(self.request_body.decode("utf-8"))
+        data = json.loads(self.request_body.decode("utf-8"))
+
+        if self.accessory_handler.accessory.category == CATEGORY_BRIDGE:
+            accessory = self.accessory_handler.accessory.accessories.get(data['aid'])
+            if not accessory:
+                raise ValueError('Accessory with aid == {} not found'.format(data['aid']))
```

Comment:
It would be nice to have test coverage for the missing `aid` case, but I can add it after https://github.com/ikalchev/HAP-python/pull/307 merges if you don't have time.

cc @ikalchev

Reply:
I could add the test myself after #307 will be merged )

Reply:
I just realized we don't have a timeout for the `async_get_snapshot`.  I'll wait to fix that until after this merges so there isn't a conflict.

Reply:
@ikalchev  

Sorry to bother, but we are a bit deadlocked with PRs since any changes at this point are likely to conflict.

https://github.com/ikalchev/HAP-python/pull/306
https://github.com/ikalchev/HAP-python/pull/307
https://github.com/ikalchev/HAP-python/pull/308
https://github.com/ikalchev/HAP-python/pull/310

Reply:
I merged all except this one. Is this still good to go?

Reply:
Yes.  As soon as it merges, I'll send a new PR to fix the async timeout issue (not a new problem)

Reply:
@bzzeke I cherry picked your changes into https://github.com/ikalchev/HAP-python/pull/311 and  added test coverage + the timeout fix, so I think this PR can be closed since your changes will merge without conflict in #311

I've re-tested the changes on 3 systems with everything integrated into a branch.

@ikalchev #311 should be ready to merge + release 3.3.0?

Reply:
Done! New release over the weekend :)

Reply:
Thank you.  Updated 3 production instances to `dev` everything and all looking good for a release!

Reply:
Thank you! So I'm closing this PR in favour of #311